aws/resource_aws_vpn_gateway_test.go

```diff
@@ -497,6 +478,14 @@ resource "aws_vpn_gateway" "test" {
 `
 
 const testAccVpnGatewayConfigChangeVPC = `
+resource "aws_vpc" "test" {
+  cidr_block = "10.1.0.0/16"
+
+  tags = {
+    Name = "terraform-testacc-vpn-gateway"
+  }
+}
+
```

Comment:
Else

```
=== CONT  TestAccAWSVpnGateway_basic
    resource_aws_vpn_gateway_test.go:119: Step 3/3 error: terraform failed: exit status 1
        
        stderr:
        
        Error: Error deleting VPC: DependencyViolation: The vpc 'vpc-00e1f817ec8e815b1' has dependencies and cannot be deleted.
        	status code: 400, request id: 75fd77ad-805e-47a5-bc10-9fd380522c7d
        
        
--- FAIL: TestAccAWSVpnGateway_basic (357.00s)
```

Reply:
This change is totally fine (while maybe `create_before_destroy` in the previous test configuration would "correctly" fix this), but this upstream issue has a little more context about the (IMO unexpected 😉 ) ordering behavior: https://github.com/hashicorp/terraform/issues/26173

Reply:
Did the order of operations change with TF 0.12?
This test must have been running fine with the "internal" version of TF used to run acceptance tests before migration to Plugin SDK v2.

Reply:
I believe so, at some point during the Terraform 0.12.x line between 0.12.7 (~ when the SDK version of core was split) and 0.12.26 (the first version the binary testing framework really supports).